src/main/java/seedu/ezdo/ui/TaskCard.java

```diff
@@ -54,11 +72,63 @@ public TaskCard(ReadOnlyTask task, int displayedIndex) {
         name.setText(task.getName().fullName);
         id.setText(displayedIndex + ". ");
         setPriority(task);
+        setStartDate(task);
+        setDueDate(task);
+        initTags(task);
+    }
+
+    // ========================= STARTDATE ============================ //
+
+    private void setStartDate(ReadOnlyTask task) {
+        Date currentDate = new Date();
         startDate.setText(task.getStartDate().value);
+        setStartDateColor(currentDate, CSS_STARTDATE_PAST_CURRENT_DATE_COLOR);
+    }
+
+    private void setStartDateColor(Date dateReference, String cssColor) {
+
+        try {
+            if (dateReference.after(DATE_FORMAT.parse(startDate.getText()))) {
+                startDate.setStyle(cssColor);
+            }
+        } catch (ParseException pe) {
+            // Do nothing as the start date is optional
+            // and cannot be parsed as Date object
+        }
+    }
+
+    // ========================= DUEDATE ============================ //
+
+    private void setDueDate(ReadOnlyTask task) {
+        Date currentDate = new Date();
+        Date dateAWeekInAdvance = createDateAWeekInAdvance();
+
         dueDate.setText(task.getDueDate().value);
-        initTags(task);
+        setDueDateColor(dateAWeekInAdvance, CSS_ABOUT_TO_DUE_COLOR);
+        setDueDateColor(currentDate, CSS_OVERDUE_COLOR);
     }
 
+    private void setDueDateColor(Date dateReference, String cssColor) {
+
+        try {
+            if (dateReference.after(DATE_FORMAT.parse(dueDate.getText()))) {
+                startDate.setStyle(null);
+                dueDate.setStyle(cssColor);
+            }
+        } catch (ParseException pe) {
+            // Do nothing as the due date is optional
+            // and cannot be parsed as Date object
+        }
+    }
+
+    private Date createDateAWeekInAdvance() {
```

Comment:
oh I just realised that's to change date color to orange if it's due in 7 days. 
no problem with this method.